tests/test-class-wp-https-detection.php

```diff
@@ -80,60 +138,102 @@ public function test_schedule_cron() {
 		$this->assertEquals(
 			array(
 				'args'     => array(),
-				'interval' => HOUR_IN_SECONDS,
-				'schedule' => 'hourly',
+				'interval' => DAY_IN_SECONDS / 2,
+				'schedule' => 'twicedaily',
 			),
 			reset( $https_check_cron[ WP_HTTPS_Detection::CRON_HOOK ] )
 		);
 	}
 
 	/**
-	 * Test update_option_https_support.
+	 * Test update_https_support_options.
 	 *
-	 * @covers WP_HTTPS_Detection::update_option_https_support()
+	 * @covers WP_HTTPS_Detection::update_https_support_options()
 	 */
-	public function test_update_option_https_support() {
-		$this->instance->update_option_https_support();
+	public function test_update_https_support_options() {
+		add_filter( 'http_response', array( $this, 'mock_successful_response' ) );
+		$this->instance->update_https_support_options();
 		$this->assertTrue( get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) );
+		remove_filter( 'http_response', array( $this, 'mock_successful_response' ) );
 
-		// There should be HTTPS support, as check_https_support() should return true.
-		add_filter( 'http_response', array( $this, 'mock_error_response' ) );
-		$this->instance->update_option_https_support();
-		$this->assertTrue( get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) );
+		// The HTTPS support option should be false, as the request for the HTTPS page failed.
+		$this->instance->update_https_support_options();
+		$this->assertFalse( get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) );
 		remove_filter( 'http_response', array( $this, 'mock_error_response' ) );
 
 		// The response is a 301, so the option value should be false.
 		add_filter( 'http_response', array( $this, 'mock_incorrect_response' ) );
-		$this->instance->update_option_https_support();
+		$this->instance->update_https_support_options();
 		$this->assertFalse( get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) );
 		remove_filter( 'http_response', array( $this, 'mock_incorrect_response' ) );
 	}
 
+	/**
+	 * Test is_currently_https.
+	 *
+	 * @covers WP_HTTPS_UI::is_currently_https()
+	 */
+	public function test_is_currently_https() {
+		// If both of these options have an HTTP URL, the method should return false.
+		update_option( 'siteurl', self::HTTP_URL );
+		update_option( 'home', self::HTTP_URL );
+		$this->assertFalse( $this->instance->is_currently_https() );
+
+		// If one of these options has an HTTP URL, the method should return false.
+		update_option( 'siteurl', self::HTTPS_URL );
+		$this->assertFalse( $this->instance->is_currently_https() );
+
+		// If both of these options have an HTTPS URL, the method should return true.
+		update_option( 'siteurl', self::HTTPS_URL );
+		update_option( 'home', self::HTTPS_URL );
+		add_filter( 'set_url_scheme', array( $this, 'convert_to_https' ) );
+		$this->assertTrue( $this->instance->is_currently_https() );
```

Comment:
@kienstra We need another set of assertions to validate having a proper manifest.

Reply:
Hi Tonya, https://github.com/xwp/pwa-wp/pull/36/commits/0fb5b7ef260d7829eafa90fc587554452fd13c12 now adds an assertion for there being an incorrect manifest.

This [assertion](https://github.com/xwp/pwa-wp/blob/0fb5b7ef260d7829eafa90fc587554452fd13c12/tests/test-class-wp-https-detection.php#L156) tested a proper manifest being output, [by adding](https://github.com/xwp/pwa-wp/blob/0fb5b7ef260d7829eafa90fc587554452fd13c12/tests/test-class-wp-https-detection.php#L154) `mock_successful_response` as a callback.